lib/ecto/repo.ex

```diff
@@ -681,4 +681,16 @@ defmodule Ecto.Repo do
 
   """
   @callback log(Ecto.LogEntry.t) :: any
+
+  @doc false
+  def transaction(adapter, repo, opts, fun) when is_function(fun, 0) do
+    adapter.transaction(repo, opts, fun)
+  end
+
+  def transaction(adapter, repo, opts, %Ecto.Multi{} = multi) do
+    wrap   = &adapter.transaction(repo, &1, &2)
```

Comment:
But I still want to pass the options to all the repo functions. So we respect something like `log: false`.